### PR TITLE
ElementHighlighter

### PR DESCRIPTION
**Pull request by kdzwinel**
element-highlighter mov

Takes element out of the page, shows overlay over the page, and puts element on the top of the overlay in the exact same place it was originally.

### CODE REVIEW

File: components/element-highlighter/src/js/animation-helper.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+function onTransitionEnd(element, callback) {
```
Comment (kdzwinel) [inline components/element-highlighter/src/js/animation-helper.js]: basically jQuery's `.once('transitionend', () => {})` + making sure that the event comes from given element and not one of it's children

File: components/element-highlighter/src/sass/default.scss [hunk 2]
```diff
@@ -0,0 +1,23 @@
+$buicEhOverlayColor: #000000;
```
Comment (kdzwinel) [inline components/element-highlighter/src/sass/default.scss]: minimal default styling - just the transparent black overlay (blur shown in demo is not part of the defaults)

File: components/template/index.html [hunk 3]
```diff
@@ -2,9 +2,15 @@
 <html lang="en">
```
Comment (kdzwinel) [inline components/template/index.html]: small improvements to component's template

File: index.html [hunk 4]
```diff
@@ -98,6 +101,7 @@ <h1 class="sg-header-primary">Library of UI components</h1>
     .buic-fc-box {
       width: 100%;
       height: 100%;
+      z-index: 1;
```
Comment (kdzwinel) [inline index.html]: small issue with flip-card demo spotted and fixed

File: index.html [hunk 5]
```diff
@@ -248,7 +252,7 @@ <h1 class="sg-header-primary">Library of UI components</h1>
             <div class="buic-carousel__slider">
               <div class="buic-carousel__slide">
                 <p class="sg-text-bit sg-text-bit--not-responsive sg-text-bit--xlarge sg-text-bit--light">
-                  1st slide
+                  Swipe &rarr;
```
Comment (kdzwinel) [inline index.html]: Helping people understand how carousel works

File: components/element-highlighter/src/js/main.js [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    placeholder.style.width = elementBcr.width + 'px';
+    placeholder.style.height = elementBcr.height + 'px';
+    placeholder.style.marginLeft = elementStyles.marginLeft;
+    placeholder.style.marginRight = elementStyles.marginRight;
+    placeholder.style.marginTop = elementStyles.marginTop;
+    placeholder.style.marginBottom = elementStyles.marginBottom;
+
+    elementParent.replaceChild(placeholder, this.element);
+
+    // crate an overlay
```
Comment (matzimowski) [inline components/element-highlighter/src/js/main.js]: typo

**Pull request merged**